batcher/aligned-batcher/src/eth/utils.rs

```diff
@@ -10,3 +16,24 @@ pub fn get_provider(eth_rpc_url: String) -> Result<Provider<Http>, anyhow::Error
         .map_err(|e| anyhow::Error::msg(format!("Failed to create provider: {}", e)))?;
     Ok(Provider::new(provider))
 }
+
+pub async fn get_batcher_signer(
+    provider: Provider<Http>,
+    ecdsa_config: ECDSAConfig,
+) -> Result<Arc<SignerMiddlewareT>, anyhow::Error> {
+    let chain_id = provider.get_chainid().await?;
+
+    let escalator = GeometricGasPrice::new(GAS_MULTIPLIER, GAS_ESCALATOR_INTERVAL, None::<u64>);
+
+    let provider = GasEscalatorMiddleware::new(provider, escalator, Frequency::PerBlock);
```

Comment:
is this needed? I think it may be some legacy code

Reply:
In theory, this increases the transaction's gas price every block. I can remove it if you want

Reply:
I would prefer to write this code ourselves, it shouldn't be so hard. We already have some of it written in go.

Reply:
@entropidelic what do you think?